### ISSUE TITLE
Incomplete support for "data-" prefix

### ISSUE THREAD

**Issue by jsampson** [bigskysoftware/htmx#19]
While the docs say that the "data-" prefix is supported for all "hx-" attributes, there appear to be several places in the code that do not honor it. This is based on reviewing the code, not running it. The unit tests don't have coverage for "data-" prefixes, and I haven't reproduced any of these issues.

* [L27](https://github.com/bigskysoftware/htmx/blob/cb8f9b6f1084848b76ebe11dea8ce1ed6306205c/src/htmx.js#L27): comment mentions "kt and data-kt prefixes"
    * should become "hx and data-hx prefixes"
* [L53](https://github.com/bigskysoftware/htmx/blob/cb8f9b6f1084848b76ebe11dea8ce1ed6306205c/src/htmx.js#L53): getClosestAttributeValue uses getRawAttribute
    * should use getAttributeValue
* [L293](https://github.com/bigskysoftware/htmx/blob/cb8f9b6f1084848b76ebe11dea8ce1ed6306205c/src/htmx.js#L293) & [L295](https://github.com/bigskysoftware/htmx/blob/cb8f9b6f1084848b76ebe11dea8ce1ed6306205c/src/htmx.js#L295): getTarget uses getRawAttribute for "hx-target"
    * should use getAttributeValue
* [L631](https://github.com/bigskysoftware/htmx/blob/cb8f9b6f1084848b76ebe11dea8ce1ed6306205c/src/htmx.js#L631): initScrollHandler uses `querySelectorAll("[hx-trigger='revealed']")`
    * could be `querySelectorAll("[hx-trigger='revealed'],[data-hx-trigger='revealed']")`
* [L801](https://github.com/bigskysoftware/htmx/blob/cb8f9b6f1084848b76ebe11dea8ce1ed6306205c/src/htmx.js#L801): getHistoryElement uses `querySelector('[hx-history-elt]')`
    * could be `querySelector("[hx-history-elt],[data-hx-history-elt]")`
* [L858](https://github.com/bigskysoftware/htmx/blob/cb8f9b6f1084848b76ebe11dea8ce1ed6306205c/src/htmx.js#L858): loadHistoryFromServer uses `querySelector('[hx-history-elt]')`
    * could be `querySelector("[hx-history-elt],[data-hx-history-elt]")`
* [L1095](https://github.com/bigskysoftware/htmx/blob/cb8f9b6f1084848b76ebe11dea8ce1ed6306205c/src/htmx.js#L1095): issueAjaxRequest uses getRawAttribute for "hx-target"
    * should use getAttributeValue

**Comment by chg20** [comment]
Good catches.

Should probably have a test in each attribute test for the "data-" prefix.

**Comment by jsampson** [comment]
Yep, that probably makes sense. I've added notes in the issue description about the likely fixes. After a bit of experimenting in the browser console It seems like the comma in selectors works as "or" just like in CSS. There might be something more elegant.

I added an [Htmx binding in Remarker](https://github.com/jsampson/remarker/blob/master/src/main/java/org/remarker/ext/Htmx.java) (my server-side rendering library that nobody uses) and decided to use the "data-" prefixed attributes on a lark. Then figured I should check how/whether they work. :)
